unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/CylinderShape/ECSCylinderShapeComponentHandler.cs

```diff
@@ -0,0 +1,64 @@
+using System.Collections;
+using System.Collections.Generic;
+using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Models;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public class ECSCylinderShapeComponentHandler : IECSComponentHandler<PBCylinderShape>
+    {
+        internal AssetPromise_PrimitiveMesh primitiveMeshPromisePrimitive;
+        internal MeshesInfo meshesInfo;
+        internal Rendereable rendereable;
+
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity) { }
+
+        public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+        {
+            if (primitiveMeshPromisePrimitive != null)
+                AssetPromiseKeeper_PrimitiveMesh.i.Forget(primitiveMeshPromisePrimitive);
+            DisposeMesh(scene);
+        }
+
+        public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBCylinderShape model)
+        {
+            Mesh generatedMesh = null;
+            if (primitiveMeshPromisePrimitive != null)
+                AssetPromiseKeeper_PrimitiveMesh.i.Forget(primitiveMeshPromisePrimitive);
+
+            PrimitiveMeshModel primitiveMeshModelModel = new PrimitiveMeshModel(PrimitiveMeshModel.Type.Cylinder);
+            primitiveMeshPromisePrimitive = new AssetPromise_PrimitiveMesh(primitiveMeshModelModel);
+            primitiveMeshPromisePrimitive.model.radiusBottom = model.RadiusBottom;
+            primitiveMeshPromisePrimitive.model.radiusTop = model.RadiusTop;
```

Comment:
why the modification of the inner `model` field of the promise?
shouldn't those modification be applied to `primitiveMeshModelModel` instead?